### PR TITLE
fix: Absolute Cart Discount Money marshaling (to list)

### PR DESCRIPTION
**Pull request by papkos**
The `manyMoney` part was copied from [`flattenProductDiscountValue`](https://github.com/labd/terraform-provider-commercetools/blob/db3a6bad9ca898d9c0b406489bea6d5919929e92/commercetools/resource_product_discount.go#L360).

The `panic` message change helped confirm that the issue indeed was that the function got a slice of `TypedMoney`, which it couldn't handle.  
At some point it might also get a `*TypedMoney`, which it also won't be able to handle (happened in our code while using the SDK). If that ever happens, we'll get a more helpful message now :) 

Fixes https://github.com/labd/terraform-provider-commercetools/issues/344

### CODE REVIEW

File: commercetools/resource_cart_discount.go [hunk 1]
```diff
@@ -483,14 +483,24 @@ func flattenCartDiscountValue(val platform.CartDiscountValue) []map[string]any {
 
 	switch v := val.(type) {
 	case platform.CartDiscountValueAbsolute:
+		manyMoney := make([]map[string]any, len(v.Money))
```
Comment (FluffyBucket) [inline commercetools/resource_cart_discount.go:486]: You did not wanted to call it `moneys`? :stuck_out_tongue:

**Review by FluffyBucket — APPROVED** [review]

**Review by rioam2 — APPROVED** [review]

**Comment by zackpollard** [comment]
Hey, thanks for the work on this, any chance we can get this merged soon? :smile:

**Comment by valerydluski** [comment]
Hello, could you tell me how soon this pull request is planned to be merged? We are really looking forward to this fix.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -47,7 +47,8 @@ jobs:
         CTP_API_URL: http://localhost:8989
         CTP_AUTH_URL: http://localhost:8989
 
-    - name: Upload to codecov
-      uses: codecov/codecov-action@v3
-      with:
-        verbose: true
+    # Don't run this for this fork
```
Comment (demeyerthom) [inline .github/workflows/tests.yml:50]: Code you please re-enable this?

File: commercetools/resource_cart_discount.go [hunk 3]
```diff
@@ -483,14 +483,24 @@ func flattenCartDiscountValue(val platform.CartDiscountValue) []map[string]any {
...
+		for i, money := range v.Money {
+			manyMoney[i] = flattenTypedMoney(money)
+		}
 		return []map[string]any{{
-			"type":  "absolute",
-			"money": flattenTypedMoney(v.Money),
+			"type":      "absolute",
+			"money":     manyMoney,
+			"permyriad": 0,
```
Comment (demeyerthom) [inline commercetools/resource_cart_discount.go:493]: I don't think the API supports `permyriad` in this case?

File: commercetools/resource_cart_discount.go [hunk 4]
```diff
@@ -483,14 +483,24 @@ func flattenCartDiscountValue(val platform.CartDiscountValue) []map[string]any {
...
+		manyMoney := make([]map[string]any, len(v.Money))
+		for i, money := range v.Money {
+			manyMoney[i] = flattenTypedMoney(money)
+		}
 		return []map[string]any{{
-			"type":  "fixed",
-			"money": flattenTypedMoney(v.Money),
+			"type":      "fixed",
+			"money":     manyMoney,
+			"permyriad": 0,
```
Comment (demeyerthom) [inline commercetools/resource_cart_discount.go:503]: I don't think the API supports permyriad in this case?

File: commercetools/resource_cart_discount_absolute_test.go [hunk 5]
```diff
@@ -0,0 +1,86 @@
+package commercetools
```
Comment (demeyerthom) [inline commercetools/resource_cart_discount_absolute_test.go:1]: Can we also add a test for the fixed discount? That way we have all cases except the `giftLineItem` covered

**Review by demeyerthom — CHANGES_REQUESTED** [review]
@papkos thanks for creating this PR!

I left some comments I hope you can look into, but overall it looks good to me!

**Comment by demeyerthom** [comment]
Merged this to v1.8.0 branch and will fix testing there

**Comment by papkos** [comment]
> Merged this to v1.8.0 branch and will fix testing there

Thank you, and sorry for not getting back to this sooner. 
Looking forward to the new release!

**Pull request merged**